homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -0,0 +1,83 @@
+"""DataUpdateCoordinator for the swiss_public_transport integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import TypedDict
+
+from opendata_transport import OpendataTransport
+from opendata_transport.exceptions import OpendataTransportError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DataConnection(TypedDict):
+    """A connection data class."""
+
+    departure: str
+    next_departure: str
+    next_on_departure: str
+    duration: str
+    platform: str
+    remaining_time: str
+    start: str
+    destination: str
+    train_number: str
+    transfers: str
+    delay: int
+
+
+class SwissPublicTransportDataUpdateCoordinator(DataUpdateCoordinator[DataConnection]):
+    """A SwissPublicTransport Data Update Coordinator."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, opendata: OpendataTransport) -> None:
+        """Initialize the SwissPublicTransport data coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=90),
+        )
+        self._opendata = opendata
+        self._remaining_time: timedelta | None = None
```

Comment:
```suggestion
    config_entry: ConfigEntry
    _remaining_time: timedelta | None = None

    def __init__(self, hass: HomeAssistant, opendata: OpendataTransport) -> None:
        """Initialize the SwissPublicTransport data coordinator."""
        super().__init__(
            hass,
            _LOGGER,
            name=DOMAIN,
            update_interval=timedelta(seconds=90),
        )
        self._opendata = opendata
```

Reply:
Wait

Reply:
I dont think you need to store the remaining time in a class attribute anymore

Reply:
Good spot, and I messed up the ordering of the attributes...